library/iFixit/Matryoshka/Memcached.php

```diff
@@ -60,16 +89,38 @@ public function decrement($key, $amount = 1, $expiration = 0) {
        $expiration);
    }
 
+   /**
+    * @template T
+    * @return T|self::MISS
+    */
    public function get($key) {
+      /** @var T|false */
       $value = $this->memcached->get($key);
+      $result = $this->memcached->getResultCode();
 
-      if ($this->memcached->getResultCode() === \Memcached::RES_NOTFOUND) {
+      if ($this->isMiss($result)) {
          return self::MISS;
       }
 
+      $error = $this->getError($result);
+      if ($error) {
+         throw $error;
```

Comment:
This is kind of surprising behavior change in a public library.

But i think it's a valid one.